agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/statsbeat/NetworkStatsbeat.java

```diff
@@ -178,48 +197,74 @@ private void sendIntervalMetric(
               telemetryClient,
               REQUEST_FAILURE_COUNT_METRIC_NAME,
               (double) local.requestFailureCount.get());
-      addCommonProperties(requestFailureCountSt, ikey, local.host);
+      addCommonProperties(requestFailureCountSt, key);
       telemetryClient.trackStatsbeatAsync(requestFailureCountSt.build());
     }
 
     double durationAvg = local.getRequestDurationAvg();
     if (durationAvg != 0) {
       StatsbeatTelemetryBuilder requestDurationSt =
           createStatsbeatTelemetry(telemetryClient, REQUEST_DURATION_METRIC_NAME, durationAvg);
-      addCommonProperties(requestDurationSt, ikey, local.host);
+      addCommonProperties(requestDurationSt, key);
       telemetryClient.trackStatsbeatAsync(requestDurationSt.build());
     }
 
     if (local.retryCount.get() != 0) {
       StatsbeatTelemetryBuilder retryCountSt =
           createStatsbeatTelemetry(
               telemetryClient, RETRY_COUNT_METRIC_NAME, (double) local.retryCount.get());
-      addCommonProperties(retryCountSt, ikey, local.host);
+      addCommonProperties(retryCountSt, key);
       telemetryClient.trackStatsbeatAsync(retryCountSt.build());
     }
 
     if (local.throttlingCount.get() != 0) {
       StatsbeatTelemetryBuilder throttleCountSt =
           createStatsbeatTelemetry(
               telemetryClient, THROTTLE_COUNT_METRIC_NAME, (double) local.throttlingCount.get());
-      addCommonProperties(throttleCountSt, ikey, local.host);
+      addCommonProperties(throttleCountSt, key);
       telemetryClient.trackStatsbeatAsync(throttleCountSt.build());
     }
 
     if (local.exceptionCount.get() != 0) {
       StatsbeatTelemetryBuilder exceptionCountSt =
           createStatsbeatTelemetry(
               telemetryClient, EXCEPTION_COUNT_METRIC_NAME, (double) local.exceptionCount.get());
-      addCommonProperties(exceptionCountSt, ikey, local.host);
+      addCommonProperties(exceptionCountSt, key);
       telemetryClient.trackStatsbeatAsync(exceptionCountSt.build());
     }
   }
 
   private static void addCommonProperties(
-      StatsbeatTelemetryBuilder telemetryBuilder, String ikey, String host) {
+      StatsbeatTelemetryBuilder telemetryBuilder, IntervalMetricsKey key) {
     telemetryBuilder.addProperty("endpoint", BREEZE_ENDPOINT);
-    telemetryBuilder.addProperty("cikey", ikey);
-    telemetryBuilder.addProperty("host", host);
+    telemetryBuilder.addProperty("cikey", key.getIkey());
+    telemetryBuilder.addProperty("host", key.getHost());
+    Object cause = key.getCause();
+    if (cause != null) {
+      // track 'statusCode' for Failure/Retry/Throttle count and track 'exceptionType' for Exception
+      // Count
+      if (cause instanceof Integer) {
+        telemetryBuilder.addProperty("statusCode", ((Integer) cause).toString());
+      } else if (cause instanceof String) {
+        telemetryBuilder.addProperty("exceptionType", (String) cause);
+      }
```

Comment:
I think I prefer the more explicit `causeFieldName` + `causeValue`, instead of the single `cause` and relying here on knowing that `Integer` == "statusCode" and `String` == "exceptionType"